app/services/policies/policies_list_service.rb

```diff
@@ -61,4 +65,25 @@ def to_h
       @sets.transform_values(&:to_a).as_json
     end
   end
+
+  HTTP_ERRORS = [HTTPClient::BadResponseError, HTTPClient::TimeoutError, HTTPClient::ConnectTimeoutError, HTTPClient::SendTimeoutError, HTTPClient::ReceiveTimeoutError, SocketError].freeze
+  private_constant :HTTP_ERRORS
+
+  def self.apicast_registry_url
+    ThreeScale.config.sandbox_proxy.apicast_registry_url
+  end
+
+  def self.fetch_policies_from_apicast
+    response = ::JSONClient.get(apicast_registry_url)
+    response.body['policies'] if response.ok? or raise_policies_list_error(response.content)
+  rescue *HTTP_ERRORS => error
+    raise_policies_list_error(error)
```

Comment:
considering that this is in a private method that is only called from `call`, I personally wouldn't rescue `*HTTP_ERRORS` to raise them as `PoliciesListServiceError`, to rescue them in the `call` of this same class. I would just let the HTTP_ERRORS be raised and in the line 15 where you are doing `rescue PoliciesListServiceError => error`, I would do `rescue PoliciesListServiceError, *HTTP_ERRORS => error`.

But let's see what @guicassolato and @macejmic think about this too 😄

Reply:
I agree with @Martouta. We are raising `PoliciesListServiceError` only to be caught by `#call`.